lib_eio/net.mli

```diff
@@ -78,10 +78,35 @@ module Sockaddr : sig
   val pp : Format.formatter -> [< t] -> unit
 end
 
+(** {2 Socket Options} *)
+
+type 'a sockopt =
```

Comment:
Putting it in a module means you can remove the `SO_` prefixes:
```suggestion
module Sock_opt = struct
  type 'a t =
    | IPV6_ONLY  : bool sockopt  (** Forbid binding an IPv6 socket to an IPv4 address *)
    | ACCEPTCONN : bool sockopt  (** Report whether socket listening is enabled *)
    | BROADCAST  : bool sockopt  (** Permit sending of broadcast messages *)
    ...
```
Not sure what to call the module. It doesn't matter much, though, as OCaml will be able to infer it from the type.